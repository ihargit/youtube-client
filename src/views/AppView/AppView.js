import Search from './search/search';
import Slider from './slider/slider';
import SliderNavigation from './slider-navigation/sliderNavigation';
import Warnings from './warnings/warnings';


export default class AppView {
  constructor() {
    this.search = Search;
    this.slider = Slider;
    this.sliderNavigation = SliderNavigation;
    this.warnings = Warnings;
  }

  init() {
    const header = document.createElement('header');
    const main = document.createElement('main');

    header.appendChild(this.search.renderSearchBar());
    main.appendChild(this.slider.renderSlider());
    main.appendChild(this.sliderNavigation.renderSliderNavigation());
    main.appendChild(this.warnings.renderNotFoundWarning());
    main.appendChild(this.warnings.renderSameRequestWarning());

    document.body.appendChild(header);
    document.body.appendChild(main);
  }

  renderVideos(videosData) {
    this.slider.addVideosToSlider(videosData);
  }

  clearSliderFromVideos() {
    this.slider.emptySlider();
  }

  changeVideosTotalInSlider(number) {
    this.slider.changeVideosTotal(number);
  }

  showSliderNavigation() {
    this.sliderNavigation.showSliderNavigation();
  }

  hideSliderNavigation() {
    return this.sliderNavigation.hideSliderNav();
  }

  updateSliderNav(position) {
    this.sliderNavigation.updateSliderNavigation(position);
  }
}
