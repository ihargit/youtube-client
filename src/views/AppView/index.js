import AppView from './AppView';
import './style/fonts.css';
import './style/normalize.css';
import './style/app-view.css';
import './style/all.min.css';

export default AppView;
