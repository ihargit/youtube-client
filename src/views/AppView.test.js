import AppView from './AppView';

describe('AppView.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should render correctly', () => {
    const context = {
      titles: [
        'Video about JS',
        'Another video ...',
        'I need another video',
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
