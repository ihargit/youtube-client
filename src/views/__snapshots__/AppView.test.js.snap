// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should render correctly 1`] = `"<ul><li>Video about JS</li><li>Another video ...</li><li>I need another video</li></ul>"`;
